jawa/util/bytecode.yaml

```diff
@@ -0,0 +1,570 @@
+arraylength:
+    op: 0xBE
+astore:
+    op: 0x3A
+    operands:
+        - ['UBYTE', 'LOCAL_INDEX']
+    can_be_wide: True
+astore_0:
+    op: 0x4B
+astore_1:
+    op: 0x4C
+astore_2:
+    op: 0x4D
+astore_3:
+    op: 0x4E
+athrow:
+    op: 0xBF
+baload:
+    op: 0x33
+bastore:
+    op: 0x54
+bipush:
+    op: 0x10
+    operands:
+        - ['UBYTE', 'LITERAL']
+caload:
+    op: 0x34
+castore:
+    op: 0x55
+checkcast:
+    op: 0xC0
+    operands:
+        - ['USHORT', 'CONSTANT_INDEX']
+d2f:
+    op: 0x90
+d2i:
+    op: 0x8E
+d2l:
+    op: 0x8F
+dadd:
+    op: 0x63
+daload:
+    op: 0x31
+dastore:
+    op: 0x52
+dcmpg:
+    op: 0x98
+dcmpl:
+    op: 0x97
+dconst_0:
+    op: 0x0E
+dconst_1:
+    op: 0x0F
+ddiv:
+    op: 0x6F
+dload:
+    op: 0x18
+    operands:
+        - ['UBYTE', 'LOCAL_INDEX']
+    can_be_wide: True
+dload_0:
+    op: 0x26
+dload_1:
+    op: 0x27
+dload_2:
+    op: 0x28
+dload_3:
+    op: 0x29
+dmul:
+    op: 0x6B
+dneg:
+    op: 0x77
+drem:
+    op: 0x73
+dreturn:
+    op: 0xAF
+dstore:
+    op: 0x39
+    operands:
+        - ['UBYTE', 'LOCAL_INDEX']
+    can_be_wide: True
+dstore_0:
+    op: 0x47
+dstore_1:
+    op: 0x48
+dstore_2:
+    op: 0x49
+dstore_3:
+    op: 0x4A
+dsub:
+    op: 0x67
+dup:
+    op: 0x59
+dup_x1:
+    op: 0x5A
+dup_x2:
+    op: 0x5B
+dup2:
+    op: 0x5C
+dup2_x1:
+    op: 0x5D
+dup2_x2:
+    op: 0x5E
+f2d:
+    op: 0x8D
+f2i:
+    op: 0x8B
+f2l:
+    op: 0x8C
+fadd:
+    op: 0x62
+faload:
+    op: 0x30
+fastore:
+    op: 0x51
+fcmpg:
+    op: 0x96
+fcmpl:
+    op: 0x95
+fconst_0:
+    op: 0x0B
+fconst_1:
+    op: 0x0C
+fconst_2:
+    op: 0x0D
+fdiv:
+    op: 0x6E
+fload:
+    op: 0x17
+    operands:
+        - ['UBYTE', 'LOCAL_INDEX']
+    can_be_wide: True
+fload_0:
+    op: 0x22
+fload_1:
+    op: 0x23
+fload_2:
+    op: 0x24
+fload_3:
+    op: 0x25
+fmul:
+    op: 0x6A
+fneg:
+    op: 0x76
+frem:
+    op: 0x72
+freturn:
+    op: 0xAE
+fstore:
+    op: 0x38
+    operands:
+        - ['UBYTE', 'LOCAL_INDEX']
+    can_be_wide: True
+fstore_0:
+    op: 0x43
+fstore_1:
+    op: 0x44
+fstore_2:
+    op: 0x45
+fstore_3:
+    op: 0x46
+fsub:
+    op: 0x66
+getfield:
+    op: 0xB4
+    operands:
+        - ['USHORT', 'CONSTANT_INDEX']
+getstatic:
+    op: 0xB2
+    operands:
+        - ['USHORT', 'CONSTANT_INDEX']
+goto:
+    op: 0xA7
+    operands:
+        - ['SHORT', 'BRANCH']
+goto_w:
+    op: 0xC8
+    operands:
+        - ['INTEGER', 'BRANCH']
+i2b:
+    op: 0x91
+i2c:
+    op: 0x92
+i2d:
+    op: 0x87
+i2f:
+    op: 0x86
+i2l:
+    op: 0x85
+i2s:
+    op: 0x93
+iadd:
+    op: 0x60
+iaload:
+    op: 0x2E
+iand:
+    op: 0x7E
+iastore:
+    op: 0x4F
+iconst_m1:
+    op: 0x02
+iconst_0:
+    op: 0x03
+iconst_1:
+    op: 0x04
+iconst_2:
+    op: 0x05
+iconst_3:
+    op: 0x06
+iconst_4:
+    op: 0x07
+iconst_5:
+    op: 0x08
+idiv:
+    op: 0x6C
+if_acmpeg:
+    op: 0xA5
+    operands:
+        - ['SHORT', 'BRANCH']
+if_acmpne:
+    op: 0xA6
+    operands:
+        - ['SHORT', 'BRANCH']
+if_icmpeg:
```

Comment:
And, same here

Reply:
Whoops, good catch! I can't think of a good way of testing correctness for this file automatically, any ideas?

Also don't bother with the JSON file,the YAML file is the source. It's only converted to JSON for distribution so we don't have to add another dependency.

Reply:
I don't really think there is a good way of doing it &mdash; you're testing data, and there isn't a good way to make sure the data is correct without having the same data in your tests (in which case the tests could be just as wrong).

On the other hand, I can think of a bad way of doing it; you could add a unit test that downloads [chapter 7 of the JVMS](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-7.html) and checks to see if every instruction listed in bytecode.yaml is also present as a word somewhere in that document.  Bad idea, but it would work :P

Good to know about the YAML/JSON thing.  I was wondering why there were two files.